tilt-dev/fsnotify
# Update go.mod and add notes on the provenance of this package

Hello @nicks,

Please review the following commits I made in branch nicks/rebase:

fb09a0d02ca744cc2242562ca0f1f046df1504c0 (2021-04-20 13:51:48 -0400)
Update go.mod and add notes on the provenance of this package

f27fd4f016e25029328fa15b4e96d008589f173e (2021-04-20 13:49:26 -0400)
windows: dynamically adjust the buffer size

17857767d84aadd250f569d4b40fe436a07cd8fe (2021-04-20 13:49:26 -0400)
oops, i forgot how to cross-compile properly on windows

2cb9c609ccd1e5de42a5b7c18e674af719da16a2 (2021-04-20 13:49:26 -0400)
windows: expose support for recursive directory watches

846b32d765503c018c1b12a4edf2cb67ccee45f6 (2021-04-20 13:49:26 -0400)
Fix debugging on macos

8a5d4abb3078e5e0a46fb9e3632d576d9398acd9 (2021-04-20 13:49:26 -0400)
kqueue: Make watcher.Close() O(n) instead of O(n^2)

7f4cf4dd2b522a984eaca51d1ccee54101d3414a (2020-04-17 14:56:12 -0700)
Fix unsafe pointer conversion (#325)
Fix the following unsafe pointer conversion found using the Go 1.14
-d=checkptr gcflags (`go1.14rc1 test -gcflags=all=-d=checkptr -v ./...`)

```
fatal error: checkptr: unsafe pointer conversion

goroutine 68 [running]:
runtime.throw(0x5833e9, 0x23)
	runtime/panic.go:1112 +0x72 fp=0xc00030fba8 sp=0xc00030fb78 pc=0x432a32
runtime.checkptrAlignment(0xc00031f018, 0x544800, 0x1)
	runtime/checkptr.go:18 +0xb7 fp=0xc00030fbd8 sp=0xc00030fba8 pc=0x4063a7
github.com/fsnotify/fsnotify.(*Watcher).readEvents(0xc0002900f0)
	github.com/fsnotify/fsnotify/inotify.go:275 +0x457 fp=0xc00031ffd8 sp=0xc00030fbd8 pc=0x519037
runtime.goexit()
	runtime/asm_amd64.s:1375 +0x1 fp=0xc00031ffe0 sp=0xc00031ffd8 pc=0x463eb1
created by github.com/fsnotify/fsnotify.NewWatcher
	github.com/fsnotify/fsnotify/inotify.go:59 +0x1a5
```

Fixes #330
45d7d09e39ef4ac08d493309fa031790c15bfe8a (2020-03-11 11:35:18 -0600)
Move example usage to the readme
may resolve #328

4bf2d1fec78374803a39307bfb8d340688f4f28e (2019-10-11 19:07:59 -0600)
Updating the FAQ section for supportability with NFS & FUSE filesystems.
closes #312 #306

dc27d9650c3ebefcf0418e4527f66a589e3f9bde (2019-10-07 20:08:56 -0600)
drop 1.8.x from Travis CI
see #310

fce5a44540c687fa4461d3b7b3bf487aa452af6a (2019-10-07 20:08:56 -0600)
2019

ecf2c4d44142daee3afd723971195dd436ea3312 (2019-10-07 20:08:56 -0600)
revise editor config
tabs only for Go

0a054074d242ce56a95584bbeeb26cb05def4086 (2019-10-07 20:08:56 -0600)
Add Go modules

ab058b44498e8b7566a799372a39d150d9ea0119 (2019-10-05 16:41:51 -0600)
readme: plan to support fanotify

924410f0f293bf12ab5bc270f535f690346d53d6 (2019-10-05 16:01:05 -0600)
Open files with CLOEXEC
Fix #272

1485a34d5d5723fea214f5710708e19a831720e4 (2019-03-12 11:14:46 -0700)
Also create epoll and pipe fds with close-on-exec (#155) (#219)
* Add the unix.O_CLOEXEC to the Pipe2 call
* Add unix.EPOLL_CLOEXEC to the Epoll call
11844c0959f6fff69ba325d097fce35bd85a8e93 (2019-03-08 10:06:45 -0700)
readme: remove AppVeyor badge
Use Travis CI for Windows #284

85de84ff31f8451e102be43b4efeed7b99ae89a0 (2019-03-08 10:04:29 -0700)
Add windows to travis matrix

2216558ce4d6e91d822580127aca3c20094230f1 (2019-03-08 10:04:29 -0700)
Fix lint issue with error doc string

9152fb1da4faee1feb08facbfe3d7fa1cf623077 (2019-03-08 10:04:29 -0700)
only run golint on latest version of go

ccc981bf80385c528a65fbfdd49bf2d8da22aa23 (2018-08-30 16:02:26 -0600)
Check if channels are closed in the example (#244)
* Check if channels are closed in the example

* Check if the channels are closed before printing

1f285c67f78ce18fae9830fcb4eaef19058bb2c6 (2018-08-30 15:20:37 -0600)
darwin tests: Exchangedata is deprecated on 10.13
#264

7b866c985edfc2c9df5663bc5b81b4c907ed5406 (2018-08-30 15:20:37 -0600)
travis: less verbose tests
TravisCI: skip tip

8ba2b192bd2e582f86863c93eaae257727d3c13d (2018-08-30 14:59:23 -0600)
[#234] t.Fatalf -> t.Errorf in go routines (#266)
The documentation for t.FailNow (which is called by t.Fatalf) states
that it should not be called in go routines outside of the test go
routine.
70cc4a11de4adb99aa24da1e7b1782da5535beaa (2018-08-30 14:03:37 -0600)
Fix TestInotifyOverflow (#265)
* Queued inotify events could have been read by the test before max_queued_events was hit
1d13583d846ea9d66dcabbfefbfb9d8e6fb05216 (2018-08-30 09:48:21 -0600)
TravisCI: updating test versions

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics
